From 4c2da5c0b1a6991e201d21cc252edc3def3e397a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:29 -0400
Subject: [tools] perf/record: Remove post_processing_offset variable

Message-id: <1408462094-14194-281-git-send-email-jolsa@redhat.com>
Patchwork-id: 88287
O-Subject: [PATCH RHEL7.1 BZ1131394 280/325] perf record: Remove post_processing_offset variable
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 7ab75cffd6a1b2195944b8522673522f09e7fcb0
Author: David Ahern <dsahern@gmail.com>
Date: Wed Nov 6 11:41:36 2013 -0700

description
===========
Duplicates the data_offset from header in the session.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index 19c4db6..15280b5 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -74,7 +74,6 @@ struct perf_record {
  bool   no_buildid;
  bool   no_buildid_cache;
  long   samples;
- off_t   post_processing_offset;
 };
 
 static int write_output(struct perf_record *rec, void *buf, size_t size)
@@ -247,13 +246,14 @@ static int process_buildids(struct perf_record *rec)
 {
  struct perf_data_file *file  = &rec->file;
  struct perf_session *session = rec->session;
+ u64 start = session->header.data_offset;
 
  u64 size = lseek(file->fd, 0, SEEK_CUR);
  if (size == 0)
   return 0;
 
- return __perf_session__process_events(session, rec->post_processing_offset,
-           size - rec->post_processing_offset,
+ return __perf_session__process_events(session, start,
+           size - start,
            size, &build_id__mark_dso_hit_ops);
 }
 
@@ -429,8 +429,6 @@ static int __cmd_record(struct perf_record *rec, int argc, const char **argv)
   goto out_delete_session;
  }
 
- rec->post_processing_offset = lseek(file->fd, 0, SEEK_CUR);
-
  machine = &session->machines.host;
 
  if (file->is_pipe) {
-- 
1.7.1